### ISSUE TITLE
hack pluggins

### ISSUE THREAD

**Issue by oceatoon** [hackmdio/hackmd#787]
Hi 
we massively love HackMD
we are developpers and have imagined upgrading HackMD with a layer connecting to an external API  that could pull information from an external DB , with autocompletion or execute processes 
2 ex : 
- referencing someone : @someone would list all the people from the external DB and add a link to our plateform inside the pad
- +@Pixies + : launch a create Organization API with  
what would be the best way to do something like this ?
is there a pluggin system ?

**Comment by SISheogorath** [comment]
No, there is currently no plugin system. Currently all changes need to be made within the code base.

Feel free to contribute :) We try to review everything as soon as possible.

**Comment by oceatoon** [comment]
Hi thanks for the quick answer
can someone just point me into the one file that handles the edit typing,
or where I could add text exception like
if(typed starts with @xxx) then launch a small autocomplete proposal list
thanks

**Comment by jackycute** [comment]
Hi @oceatoon 
Please take a look at https://github.com/hackmdio/hackmd/blob/master/public/js/index.js#L3198

**Comment by SISheogorath** [comment]
@oceatoon Any progress on this? Otherwise I would like to close it.

**Comment by pirate** [comment]
It's not an "official solution", but you can hack together some basic plugin-style behavior using the codimd-cli: https://github.com/hackmdio/codimd-cli
